plugins/module_utils/foreman_helper.py

```diff
@@ -26,6 +26,34 @@
     APYPIE_IMP_ERR = traceback.format_exc()
 
 
+def _entity_spec_helper(spec):
+    """Extend an entity spec by adding entries for all alt_ and flat_names.
+    Extract ansible compatible argument_spec on the way.
+    """
+    entity_spec = {'id': {}}
+    argument_spec = {}
+    for key, value in spec.items():
+        if 'alt_names' in value:
```

Comment:
Can you elaborate on the usecase here? As your example doesn't use it ;)

Reply:
At some point, you said we get aliases almost for free. This is the implementation for it. The name 'aliases' however had already been taken by ansible.
A usecase would be that we wanted to use a different name for something than foreman, e.g. 'from_ip' instead of 'from'.

Reply:
I *do* remember myself saying (or rather writing) that, yeah. But I think I actually meant Ansible aliases.

Let's see what I can remember from that conversation:
* [Foreman uses `from` as the name for the start IP of a Subnet](https://theforeman.org/api/1.22/apidoc/v2/subnets/create.html)
* `nailgun` uses `from_` internally, as `from` is a reserved word in Python (https://github.com/SatelliteQE/nailgun/blob/60a38229de3f2e7edb076f470ca35f18c4466afb/nailgun/entities.py#L7124)
* we use `from_ip` which does sound nicer, but my brain likes to have our and Foremans params to match (if possible)

If we're going to rename our param to `from`, the following spec should do fine:
```python
'from': {'aliases': ['from_ip']}
```
and if we keep `from_ip`, we can use:
```python
'from_ip': {'aliases': ['from'], 'flat_name': 'from'}
```

In both cases, `alt_names` aren't needed?
Please note, I wrote the `dict` directly, as you cannot write `dict(from=something)` without getting a `SyntaxError`.

The only case where I'd need `alt_names` as you implemented them is when I'd want to pass both `from` and `from_ip` to Foreman, but I don't want/need that.

Reply:
FWIW, here is my original statement: https://github.com/theforeman/foreman-ansible-modules/pull/354#discussion_r305461148

And I'm kinda sure I meant Ansible aliases ;-)

Reply:
I can see your point now, and will remove the unneeded complexity here.